x-pack/plugins/index_management/public/application/components/component_templates/component_template_wizard/component_template_edit/mappings_datastreams_rollover_modal.tsx

```diff
@@ -0,0 +1,116 @@
+
+interface Props {
+  componentTemplatename: string;
+  dataStreams: string[];
+  onClose: () => void;
+  api: ReturnType<typeof useComponentTemplatesContext>['api'];
+}
+
+export const MappingsDatastreamRolloverModal: React.FunctionComponent<Props> = ({
+  componentTemplatename,
+  dataStreams,
+  onClose,
+  api,
+}) => {
+  const [error, setError] = useState<any>();
+  const [isLoading, setIsLoading] = useState(false);
+
+  const onConfirm = useCallback(() => {
+    async function confirm() {
+      try {
+        setIsLoading(true);
+        for (const dataStream of dataStreams) {
+          await api.postDataStreamRollover(dataStream);
+        }
+        await onClose();
+      } catch (err) {
+        setError(err);
+      } finally {
+        setIsLoading(false);
+      }
+    }
+
+    confirm();
+  }, [api, onClose, dataStreams]);
+
+  return (
+    <EuiConfirmModal
+      data-test-subj="mappingDatastreamRolloverModal"
+      isLoading={isLoading}
+      title={
+        <FormattedMessage
+          id="xpack.idxMgmt.componentTemplateEdit.mappingRolloverModalTitle"
+          defaultMessage="Apply mappings now and rollover?"
+        />
+      }
+      onCancel={onClose}
+      onConfirm={onConfirm}
+      cancelButtonText={
+        <FormattedMessage
+          id="xpack.idxMgmt.componentTemplateMappingsRollover.cancelButton"
+          defaultMessage="Apply on next rollover"
+        />
+      }
+      confirmButtonText={
+        <FormattedMessage
+          id="xpack.idxMgmt.componentTemplateMappingsRollover.confirmButtom"
+          defaultMessage="Apply now and rollover"
+        />
+      }
+    >
+      {error && (
+        <>
+          <EuiCallOut
+            title={
+              <FormattedMessage
+                id="xpack.idxMgmt.componentTemplateMappingsRollover.saveError"
+                defaultMessage="Unable to apply rollover"
+              />
+            }
+            color="danger"
+            iconType="alert"
+            data-test-subj="applyMappingsRolloverError"
+          >
+            <div>{error.message || error.statusText}</div>
+          </EuiCallOut>
+          <EuiSpacer size="m" />
+        </>
+      )}
+      <EuiText>
+        <FormattedMessage
+          id="xpack.idxMgmt.componentTemplateMappingsRollover.modalDescription"
+          defaultMessage="New mappings for the {templateName} component template require a rollover for the following data streams: {datastreams} You can apply the new mappings to incoming data now and force a rollover, or wait until the next rollover. Rollover timing is defined by your index lifecycle policy."
+          values={{
+            templateName: <EuiCode>{componentTemplatename}</EuiCode>,
+            datastreams: (
+              <>
+                <EuiSpacer size="m" />
+                <ul>
+                  {dataStreams.map((datastream) => (
+                    <li>
+                      <EuiCode>{datastream}</EuiCode>
+                      <br />
```

Comment:
do we need this `br`? It seems to me that with the `li` would be enough

Reply:
You are correct here the `br` seems useless